test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/BundleEdgeCaseTests.cs

```diff
@@ -91,5 +96,133 @@ public async Task GivenABundleWithConditionalUpdateByReference_WhenExecutedWithM
+                Assert.True(firstPatientResponse.Response.IsSuccessStatusCode, "First patient ingestion did not complete as expected.");
+
+                patientsUserForInitialization[i] = firstPatientResponse.Resource.ToResourceElement().ToPoco<Patient>();
+            }
+
+            // 2 - Compose multiple bundles that will run in parallel.
+            List<Bundle> bundles = new List<Bundle>();
+            for (int i = 0; i < numberOfParallelBundles; i++)
+            {
+                Bundle bundle = new Bundle() { Type = BundleType.Batch };
+                for (int j = 0; j < numberOfPatientsPerBundle; j++)
+                {
+                    // Create Patient clones in memory.
+                    Patient tempPatient = Clone(patientsUserForInitialization[j]);
+
+                    EntryComponent entryComponent = CreateEntryComponent(tempPatient);
+
+                    bundle.Entry.Add(entryComponent);
+                }
+
+                bundles.Add(bundle);
+            }
+
+            // 3 - Post bundles in parallel.
+            List<Task<Client.FhirResponse<Bundle>>> bundleTasks = new List<Task<Client.FhirResponse<Bundle>>>();
+            foreach (var bundle in bundles)
+            {
+                bundleTasks.Add(_client.PostBundleAsync(bundle, new Client.FhirBundleOptions(), cancellationToken));
+            }
+
+            Task.WaitAll(bundleTasks.ToArray());
+
+            // 4 - Validate the response of every bundle.
+            int validatedResources = 0;
+            HashSet<string> uniquePatientIds = new HashSet<string>();
+            foreach (Task<Client.FhirResponse<Bundle>> task in bundleTasks)
+            {
+                Client.FhirResponse<Bundle> fhirResponse = task.Result;
+
+                foreach (EntryComponent item in fhirResponse.Resource.Entry)
+                {
+                    validatedResources++;
+                    string status = item.Response.Status;
+                    if (status != "200" && status != "201")
+                    {
+                        Assert.Fail($"Patient '{item.Resource.Id}' returned HTTP status code {status}.");
+                    }
+
+                    if (!uniquePatientIds.Contains(item.Resource.Id))
+                    {
+                        uniquePatientIds.Add(item.Resource.Id);
+                    }
+                }
+            }
+
+            Assert.True(
+                validatedResources == totalNumberOfExpectedPatients,
+                $"Number of expected patients ({totalNumberOfExpectedPatients}) is different than the final number of validated resources ({validatedResources}).");
+
+            Assert.True(
+                uniquePatientIds.Count == numberOfPatientsPerBundle,
+                $"Total mumber of created patients ({numberOfPatientsPerBundle}) is different than the number ingested during the test {uniquePatientIds}.");
```

Comment:
Can you add an "Assert" for the final version of the resource? which is the expected output after the parallel operations?